lib/time/time.go

```diff
@@ -129,7 +147,11 @@ func fromTimestamp(thread *starlark.Thread, _ *starlark.Builtin, args starlark.T
 }
 
 func now(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	return Time(NowFunc()), nil
+	nowFunc := Now(thread)
```

Comment:
It occurred to me that it might be important for deterministic applications to be able to prevent access to the clock entirely, not just fix it at some particular time. If so, then the signature of the per-thread time function should be func() (time.Time, error), and the logic here should return a Starlark error if the per-thread now func returns an error, or if the NowFunc var is nil (instead of panicking).  What do you think?

Reply:
I think this is an excellent observation! I'll push an update in a minute.